### DISCUSSION TITLE
Filament runout sensor is not working regardless of the firmware

### DISCUSSION THREAD

**Discussion by vali20** [Jyers/Marlin#578] [Q&A] [answered]
Hello!
I have a very strange problem with filament runout sensor with your firmware, in fact with any firmware, including the original one, made by Creality.

If I activate the sensor in the firmware and make a short between the PA pin and GND, the printing stops immediately in the position where the hotend is found, does not unload filament or park and remains attached to the printed part. The LCD freezes completely, I can't select anything.
When I disconnect the short, the printer seems to reboot (the progressbar moves like booting), it asks me if I want to resume ther printing, and when I confirm, starts the printing from the begining, it does not resume from the last position.

With the original Creality firmware, it unloads the filament a couple of minutes after starting the printing, regardless of whether the sensor detects the filament or not and stops printing. But the hotend is parked and the filament is unloaded.

So, what do you think, could the motherboard be faulty or I don't know how to configure it?

Thanks!

**Comment by gothcha** [comment, marked as answer]
It sounds to me that when you add the short to PA4 you are actually shorting 3.3V to ground!! This is why the unit reboots, very dangerous. I suspect either incorrect wiring or that there is a short between PA4 pin and the Vcc pin which will be preventing the sensor operation. If you have a multimeter you could check this with power off. Not a firmware issue.  Hope this helps.

**Reply by gothcha** [reply]
Vcc is 3.3V

**Reply by Jyers** [reply]
I have also heard of this issue when the pins are reversed.

**Reply by gothcha** [reply]
PA4 is on the left & GND in the centre, use these for the switch. Viewed from top front

**Comment by vali20** [comment]
I came back with feedback. You were both right. The original sensor or cable is wrong, it has inverted outer pins. Before connecting the sensor, I checked some schemes on the net, including on the original Creality website. I was also fooled by the fact that the VCC wire was drawn yellow, while PA4 was drawn red.
Finally, thank you both for the advice and especially to Jyers for this wonderful project.
